pilot/helpers/AgentConvo.py

```diff
@@ -179,7 +179,8 @@ def load_branch(self, branch_name, reload_files=True):
             self.replace_files()
 
     def replace_files(self):
-        files = self.agent.project.get_all_coded_files()
+        relevant_files = getattr(self.agent, 'relevant_files', None)
```

Comment:
Keeping track of relevant files is per agent. If the agent doesn't keep track of files, there's no filtering.

Currently only Developer keeps track of relevant files. For example, CodeMonkey doesn't, but it's getting the AgentConvo from the Developer which does, so it also benefits from the filtering.